### ISSUE TITLE
Editor <-> Debug window viewport inconsistency

### ISSUE THREAD

**Issue by hugeblank** [godotengine/godot#63340]
### Godot version

3.4.4.stable

### System information

MacOS Monterey, Intel Iris Plus Graphics 655 1536 MB, tested on GLES2/3

### Issue description

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/16393543/180598957-02eb8613-0e52-45c2-acc9-3a294f0601d9.png">

In the editor the line and Node2D circle clearly shouldn't overlap. In the debug window the line renders as I'd expect, but the node does not. The circle I'm drawing to render the Node2D gets cut off by the end of the screen, but in the editor that's clearly impossible.

I'd expect that the marked viewport lines in the editor indicate the edge of the screen. The line2D shows that for it, that definitely is the edge of the screen, but the Node2D can only occupy the top left quarter of the viewport in order to render the way I'd expect it to. Note that this happens independent of having Hidpi mode on or off. Similarly stretch settings are not a factor. I've tried using those to fix this issue and got nowhere.

### Steps to reproduce

1. Create a Line2D element along the width and height of the marked viewport edge in the editor. We'll use this as a metric for the edge of the screen, since we can't trust the viewport lines in the editor.
2. Create a Node2D element. Attach a script that renders a circle over the position of the node.
3. Launch the debug window. Move the node towards the middle of the viewport (512, 300). Tab back to the window to see where it is.

**Comment by Zireael07** [comment]
You probably have an error in your circle drawing script. Global vs local position mixup likely.

**Comment by hugeblank** [comment]
I don't think I do. At least not from what I can tell. The position of the root node is at 0, 0, and both the line and the node are attached to that root. Additionally, printing out the global and local positions show the same values.

**Comment by Calinou** [comment]
@hugeblank Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

**Comment by kleonc** [comment]
@hugeblank By looking at the image you provided it looks like you're doing `draw_circle(position, ...)` instead of `draw_circle(Vector2(0, 0), ...)`. Of course it's just a guess (please don't make people guess and provide all the relevant information (like the code you're talking about)).

Why zero vector is correct? Each `Node2D`'s position within its own local coordinate space is always a zero vector. `Node2D.position` property is relative to the parent, not to the node itself. And `_draw_XXX` methods work in the local space (unless there's some custom transform applied with `draw_set_transform`/`draw_set_transform_matrix`).

Nonetheless, looking at the docs of e.g. `draw_circle` I don't see anything mentioning the locality of its `position` parameter. I think there's an unspoken rule in the Godot's docs saying something like _"it's local if not specified otherwise"_ so most users probably just assume/understand that a local position is expected. However, it would be better if this would be stated explicitly (otherwise it's hard to be sure (or even suppose) what's expected, especially in case of the newcomers).

So I'd say the docs for `CanvasItem.draw_XXX` methods need to be improved in that aspect.

**Comment by hugeblank** [comment]
@kleonc Yep, I did use the nodes position erroneously. Thanks so much for clarifying. I also do think the documentation should have that clarified.
